File: Python3/프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    N, M = len(triangle), len(triangle[-1])
    
    comp_a, comp_b = None, None
    for i in range(1, N):
        for j in range(len(triangle[i])):
            comp_a = triangle[i - 1][j - 1] if 0 <= j - 1 else 0
            comp_b = triangle[i - 1][j] if j < len(triangle[i - 1]) else 0
            triangle[i][j] += max(comp_a, comp_b)
    
    return max(triangle[-1])

File: C++/프로그래머스/3/43105. 정수 삼각형/정수 삼각형.cpp
#include <algorithm>
#include <vector>

using namespace std;

int solution(vector<vector<int>> triangle) {
    int compA, compB;
    int answer = 0;
    for(int i = 1; i < triangle.size(); i++) {
        for(int j = 0; j < triangle[i].size(); j++) {
            compA = 0 <= j - 1 ? triangle[i - 1][j - 1] : 0;
            compB = j < triangle[i - 1].size() ? triangle[i - 1][j] : 0;
            triangle[i][j] += max(compA, compB);
            if(i == triangle.size() - 1) answer = max(answer, triangle[i][j]);
        }
    }
    return answer;
}